### ISSUE TITLE
Support default Debezium numeric encoding

### ISSUE THREAD

**Issue by sjwiesman** [MaterializeInc/materialize#17155]
### Feature request

By default, Debezium encodes numerical as a Java BigDemical. Materialize should natively support decoding that format into a numeric. 

Internal slack https://materializeinc.slack.com/archives/C02FWJ94HME/p1673649667015269

**Comment by benesch** [comment]
cc @aljoscha @uce @umanwizard

**Comment by umanwizard** [comment]
Some thoughts:

It's straightforward to implement the basic decoding functionality with a new schema variant called `VariableScaleDecimal` or similar.

However, we then need to decide what to do about schema resolution. For example: if a variable-scale Numeric type is narrowed to a fixed-scale decimal, will Debezium start emitting a `Fixed` instead? Avro does not specify a way to resolve between `Record` and `Fixed`; so technically in that case resolution should fail, even though there's an obvious way to interpret what's going on. (Also, schema registry is by default configured not to accept incompatible versions of an existing schema, so it's entirely possible that it will fail. We should spin up Debezium, alter a table in Postgres, and see what happens).

So, basically there is a philosophical question: should we strictly follow the Avro spec and only take into account resolved types when doing schema resolution, or should we try to interpret what the user meant in as many cases as possible? I think we might already be doing the latter in some cases (I need to check)

**Comment by benesch** [comment]
Updated the issue description to fold in new learnings from #19756.

**Comment by chuck-alt-delete** [comment]
Another user is [running into this issue](https://materializeinc.slack.com/archives/C05JLFG7NAV/p1689981303229139), this time coming from a MySQL table. From what I can tell, the precisions in the table definition are all explicitly defined.
